third_party/2.7/dateutil/parser.pyi

```diff
@@ -0,0 +1,37 @@
+from typing import List, Tuple, Optional, Callable, Union, IO, Any
+from datetime import datetime
+
+__all__ = ...  # type: List[str]
+
+
+class parserinfo(object):
+    JUMP = ... # type: List[str]
+    WEEKDAYS = ... # type: List[Tuple[str, str]]
+    MONTHS = ... # type: List[Tuple[str, str]]
+    HMS = ... # type: List[Tuple[str, str, str]]
+    AMPM = ... # type: List[Tuple[str, str]]
+    UTCZONE = ... # type: List[str]
+    PERTAIN = ... # type: List[str]
+    TZOFFSET = ... # type: Dict[str, int]
+
+    def __init__(self, dayfirst: bool=..., yearfirst: bool=...) -> None: ...
+    def jump(self, name: str) -> bool: ...
+    def weekday(self, name: str) -> Union[int, None]: ...
+    def month(self, name: str) -> Union[int, None]: ...
+    def hms(self, name: str) -> Union[int, None]: ...
+    def ampm(self, name: str) -> Union[int, None]: ...
+    def pertain(self, name: str) -> bool: ...
+    def utczone(self, name: str) -> bool: ...
+    def tzoffset(self, name: str) -> Union[int, None]: ...
+    def convertyear(self, year: int) -> int: ...
+    def validate(self, res: datetime) -> bool: ...
+
+class parser(object):
+    def __init__(self, info: Optional['parserinfo']) -> None: ...
+    def parse(self, timestr: str, default: Optional[datetime],
```

Comment:
So, incorporating Paul's ideas into a new pull request would be apropos?

Skip


On Tue, Jul 5, 2016 at 11:19 PM, Guido van Rossum <notifications@github.com>
wrote:

> In third_party/2.7/dateutil/parser.pyi
> <https://github.com/python/typeshed/pull/290#discussion_r69673372>:
>
> > +
> > +    def __init__(self, dayfirst: bool=..., yearfirst: bool=...) -> None: ...
> > +    def jump(self, name: str) -> bool: ...
> > +    def weekday(self, name: str) -> Union[int, None]: ...
> > +    def month(self, name: str) -> Union[int, None]: ...
> > +    def hms(self, name: str) -> Union[int, None]: ...
> > +    def ampm(self, name: str) -> Union[int, None]: ...
> > +    def pertain(self, name: str) -> bool: ...
> > +    def utczone(self, name: str) -> bool: ...
> > +    def tzoffset(self, name: str) -> Union[int, None]: ...
> > +    def convertyear(self, year: int) -> int: ...
> > +    def validate(self, res: datetime) -> bool: ...
> > +
> > +class parser(object):
> > +    def __init__(self, info: Optional['parserinfo']) -> None: ...
> > +    def parse(self, timestr: str, default: Optional[datetime],
>
> Yes, like everyone else I'm having a hard time keeping my strings
> straight. :-)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/python/typeshed/pull/290/files/048c0d962b3e4280201db6fbf23bc0409858c8a0#r69673372>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADRNoYGZ5ezoDArTEham9M72f-O9Ddwkks5qSyy3gaJpZM4IzqWA>
> .
>